### PR TITLE
Add WHMCS OAuth2 provider

### LINKED ISSUE #397: Provider Request » WHMCS
Hello! I'm requesting a provider for WHMCS because I don't really know how that OpenID is working.
Here is the WHMCS OpenID documentation : https://docs.whmcs.com/OpenID_Connect_Developer_Guide
Thank you.

### PR DESCRIPTION
**Pull request by Max13**
Resolves #397

### CODE REVIEW

File: src/Whmcs/Provider.php [hunk 1]
```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace SocialiteProviders\Whmcs;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * https://docs.whmcs.com/OpenID_Connect_Developer_Guide
+ * Class Provider.
```
Comment (atymic) [inline src/Whmcs/Provider.php:10]: Can remove this line.

File: src/Whmcs/Provider.php [hunk 2]
```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace SocialiteProviders\Whmcs;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * https://docs.whmcs.com/OpenID_Connect_Developer_Guide
```
Comment (atymic) [inline src/Whmcs/Provider.php:9]:
```suggestion
 * @see https://docs.whmcs.com/OpenID_Connect_Developer_Guide
```
Awesome idea to add this!

File: src/Whmcs/Provider.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
+<?php
+
```
Comment (atymic) [inline src/Whmcs/Provider.php:2]: strict types please :)

File: src/Whmcs/Provider.php [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+ * Class Provider.
+ */
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'WHMCS';
+
+    /**
```
Comment (atymic) [inline src/Whmcs/Provider.php:19]: Can remove this comment as well

File: src/Whmcs/Provider.php (hunk 3 above)
Comment (Max13) [reply src/Whmcs/Provider.php:2]: Sorry, I don't understand which types we are talking about

File: src/Whmcs/Provider.php (hunk 4 above)
Comment (Max13) [reply src/Whmcs/Provider.php:19]: ?

**Pull request merged**